x-pack/plugins/fleet/public/components/agent_enrollment_flyout/use_get_agent_incoming_data.tsx

```diff
@@ -49,30 +33,81 @@ export const useGetAgentIncomingData = (
 
   let href;
   let text;
+  let message;
+
   if (!installedPolicy) {
     href = '';
     text = '';
+    message = '';
   }
 
   if (installedPolicy?.name === 'apm') {
     href = getAbsolutePath('/app/home#/tutorial/apm');
     text = i18n.translate('xpack.fleet.confirmIncomingData.installApmAgentButtonText', {
       defaultMessage: 'Install APM Agent',
     });
+    message = i18n.translate('xpack.fleet.confirmIncomingData.APMsubtitle', {
+      defaultMessage:
+        'Next, install APM agents on your hosts to collect data from your applications and services.',
+    });
   } else {
     href = getHref('integration_details_assets', {
       pkgkey: `${installedPolicy?.name}-${installedPolicy?.version}`,
     });
     text = i18n.translate('xpack.fleet.confirmIncomingData.viewDataAssetsButtonText', {
       defaultMessage: 'View assets',
     });
+    message = i18n.translate('xpack.fleet.confirmIncomingData.subtitle', {
+      defaultMessage:
+        'Next, analyze your data using our integration assets such as curated views, dashboards and more.',
+    });
   }
   const linkButton = { href, text };
 
   return {
     enrolledAgents,
     numAgentsWithData,
-    isLoading,
     linkButton,
+    message,
   };
 };
+
+/**
+ * Hook for polling incoming data for the selected agent policy.
+ * @param agentIds
+ * @returns incomingData, isLoading
+ */
+const POLLING_INTERVAL_MS = 5 * 1000; // 5 sec
+
+export const usePollingIncomingData = (agentsIds: string[]) => {
+  const timeout = useRef<number | undefined>(undefined);
+  const [incomingData, setIncomingData] = useState<IncomingDataList[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    let isAborted = false;
+
+    const poll = () => {
+      timeout.current = window.setTimeout(async () => {
+        const { data } = await sendGetAgentIncomingData({ agentsIds });
+
+        if (data?.items) {
+          setIncomingData(data?.items);
+          setIsLoading(false);
+        }
+        if (!isAborted) {
+          poll();
+        }
+      }, POLLING_INTERVAL_MS);
+    };
+
+    poll();
+    if (isAborted || incomingData.length > 0) clearTimeout(timeout.current);
+
+    return () => {
+      isAborted = true;
```

Comment:
I'm not sure when this callback function is executed, but in my local I see `isAborted` set to `true` right after the first invocation of `poll()`.

Reply:
this might be the problematic logic:

```
setIncomingData(data?.items);
...
if (incomingData.length > 0) clearTimeout();
```

In my local, the `data?agentsIds` request comes back with the agent id inside `items` with `data: false` at first, I think the timeout shouldn't be cleared in that case.

```
{
items: [
{
cb674401-fd99-4c19-8e24-4fe4604ec4b6: {
data: false
}
}
]
}
```